components/script/textinput.rs

```diff
@@ -442,12 +442,24 @@ impl<T: ClipboardProvider> TextInput<T> {
         let target_line: isize = self.edit_point.line as isize + adjust;
 
         if target_line < 0 {
-            self.edit_point.index = 0;
             self.edit_point.line = 0;
+            self.edit_point.index = 0;
+            if let Some(_) = self.selection_origin {
+                if self.selection_direction == SelectionDirection::None ||
+                    self.selection_direction == SelectionDirection::Forward
+                {
+                    self.selection_origin = Some(TextPoint { line: 0, index: 0 })
+                }
+            }
             return;
         } else if target_line as usize >= self.lines.len() {
             self.edit_point.line = self.lines.len() - 1;
             self.edit_point.index = self.current_line_length();
+            if let Some(_) = self.selection_origin {
```

Comment:
If we don't use the value inside `Some`, maybe we can just check with `is_some()`?

```suggestion
            if self.selection_origin.is_some() {
```

Same for the previous `if let Some(_)`.

Reply:
Right, fixed it